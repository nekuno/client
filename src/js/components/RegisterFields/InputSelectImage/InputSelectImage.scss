@import '../../../../scss/_partials/colors';

.input-select-image {
  margin-bottom: 50px;
  text-align: center;

  .suggested-item {
    cursor: pointer;
    display: inline-block;
    height: 128px;
    margin: 8px 24px;
    vertical-align: top;
    width: 100px;

    .suggested-image-wrapper {
      background: transparent;
      border: 1px solid $gray-border;
      box-shadow: 0 0 5px $gray-border;
      border-radius: 50%;
      -moz-border-radius: 50%;
      -webkit-border-radius: 50%;
      display: inline-block;
      height: 98px;
      position: relative;
      overflow: hidden;
      width: 98px;

      .suggested-image {
        border-radius: 50%;
        -moz-border-radius: 50%;
        -webkit-border-radius: 50%;
        height: 84px;
        left: 7px;
        position: absolute;
        top: 7px;
        width: 84px;

        img {
          border-radius: 50%;
          -moz-border-radius: 50%;
          -webkit-border-radius: 50%;
          width: 84px;
        }
      }
      .selected-item {
        background: $overlay-image;
        border-radius: 50%;
        -moz-border-radius: 50%;
        -webkit-border-radius: 50%;
        height: 84px;
        left: 7px;
        position: absolute;
        text-align: center;
        top: 7px;
        width: 84px;
        .icon-check {
          color: $white-background;
          font-size: 35px;
          line-height: 84px;

        }
      }
    }

    .text {
      color: $gray-secondary-text;
      max-height: 40px;
      overflow: hidden;
      text-align: center;

      &.selected {
        color: $gray-main-text;
      }
    }
  }
}