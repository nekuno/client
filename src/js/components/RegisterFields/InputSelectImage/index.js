import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Input from '../../ui/Input/';
import styles from './InputSelectImage.scss';

export default class InputSelectImage extends Component {

    static propTypes = {
        placeholder    : PropTypes.string,
        options        : PropTypes.array,
        onChangeHandler: PropTypes.func,
        onClickHandler : PropTypes.func
    };

    constructor(props) {
        super(props);

        this.handleChange = this.handleChange.bind(this);
        this.handleClick = this.handleClick.bind(this);

        this.state = {
            selected: [],
            suggested: props.options
        }
    }

    handleChange(text) {
        const {options} = this.props;

        if (text) {
            this.setState({
                suggested: options.filter(
                    option => option.text.toLowerCase().indexOf(text.toLowerCase()) !== -1
                )
            });
        } else {
            this.setState({suggested: options});
        }
        if (this.props.onChangeHandler) {
            this.props.onChangeHandler(text);
        }
    }

    handleClick(id) {
        const {options} = this.props;
        const {selected} = this.state;
        let newSelected = selected.slice(0);
        const index = selected.findIndex(option => option.id === id);

        if (index !== -1) {
            newSelected.splice(index, 1);
        } else {
            newSelected.push(options.find(option => option.id === id));
        }
        this.setState({selected: newSelected});

        this.refs["input"].clearValue();

        if (this.props.onClickHandler) {
            this.props.onClickHandler(newSelected.map(option => option.id));
        }
    }

    render() {
        const {placeholder} = this.props;
        const {selected, suggested} = this.state;

        return (
            <div className={styles.inputSelectImage}>
                <Input ref="input" placeholder={placeholder} searchIcon={true} size={'small'} onChange={this.handleChange} doNotScroll={true}/>

                {suggested.map((item, index) =>
                    <div key={index} className={styles.suggestedItem}>
                        <div className={styles.suggestedImageWrapper} onClick={this.handleClick.bind(this, item.id)}>
                            <div className={styles.suggestedImage}>
                                <img className={styles.picture} src={item.picture} />
                            </div>
                            {selected.some(selectedItem => selectedItem.id === item.id) ?
                                <div className={styles.selectedItem}>
                                    <span className={styles.iconCheck + ' icon icon-check'}/>
                                </div>
                                : null
                            }
                        </div>
                        {selected.some(selectedItem => selectedItem.id === item.id) ?
                            <div className={styles.text + ' ' + styles.selected + ' small'}>{item.text}</div>
                            :
                            <div className={styles.text + ' small'}>{item.text}</div>
                        }
                    </div>
                )}
            </div>
        );
    }
}