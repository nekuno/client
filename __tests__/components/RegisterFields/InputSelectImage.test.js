import React from 'react';
import InputSelectImage from '../../../src/js/components/RegisterFields/InputSelectImage/InputSelectImage.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test InputSelectImage component', () => {
    const mockFn = jest.fn();
    const options = [
        {
            id: "lorem",
            text: "Lorem",
            image: "http://via.placeholder.com/300x300"
        },
        {
            id: "ipsum",
            text: "Ipsum",
            image: "http://via.placeholder.com/300x300"
        },
        {
            id: "lorem-ipsum",
            text: "Lorem ipsum",
            image: "http://via.placeholder.com/300x300"
        },
        {
            id: "sed-ut-perspiciatis",
            text: "Sed ut perspiciatis",
            image: "http://via.placeholder.com/300x300"
        },
        {
            id: "perspiciatis",
            text: "perspiciatis",
            image: "http://via.placeholder.com/300x300"
        },
    ];

    const inputSelectImage = shallow(<InputSelectImage options={options} placeholder={"Foo"} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(inputSelectImage).toBeDefined();
    });
    it('should render correctly', () => {
        expect(inputSelectImage).toMatchSnapshot();
    });
});
