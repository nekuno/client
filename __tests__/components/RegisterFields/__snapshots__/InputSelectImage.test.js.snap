// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputSelectImage component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputSelectImage
    onClickHandler={[MockFunction]}
    options={
      Array [
        Object {
          "id": "lorem",
          "image": "http://via.placeholder.com/300x300",
          "text": "Lorem",
        },
        Object {
          "id": "ipsum",
          "image": "http://via.placeholder.com/300x300",
          "text": "Ipsum",
        },
        Object {
          "id": "lorem-ipsum",
          "image": "http://via.placeholder.com/300x300",
          "text": "Lorem ipsum",
        },
        Object {
          "id": "sed-ut-perspiciatis",
          "image": "http://via.placeholder.com/300x300",
          "text": "Sed ut perspiciatis",
        },
        Object {
          "id": "perspiciatis",
          "image": "http://via.placeholder.com/300x300",
          "text": "perspiciatis",
        },
      ]
    }
    placeholder="Foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Input
          doNotScroll={true}
          onChange={[Function]}
          placeholder="Foo"
          searchIcon={true}
          size="small"
        />,
        Array [
          <div
            className="suggestedItem"
          >
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>
            <div
              className="text small"
            >
              Lorem
            </div>
          </div>,
          <div
            className="suggestedItem"
          >
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>
            <div
              className="text small"
            >
              Ipsum
            </div>
          </div>,
          <div
            className="suggestedItem"
          >
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>
            <div
              className="text small"
            >
              Lorem ipsum
            </div>
          </div>,
          <div
            className="suggestedItem"
          >
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>
            <div
              className="text small"
            >
              Sed ut perspiciatis
            </div>
          </div>,
          <div
            className="suggestedItem"
          >
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>
            <div
              className="text small"
            >
              perspiciatis
            </div>
          </div>,
        ],
      ],
      "className": "inputSelectImage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "doNotScroll": true,
          "onChange": [Function],
          "placeholder": "Foo",
          "searchIcon": true,
          "size": "small",
        },
        "ref": "input",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>,
            <div
              className="text small"
            >
              Lorem
            </div>,
          ],
          "className": "suggestedItem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>,
                null,
              ],
              "className": "suggestedImageWrapper",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />,
                  "className": "suggestedImage",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "image",
                    "src": "http://via.placeholder.com/300x300",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem",
              "className": "text small",
            },
            "ref": null,
            "rendered": "Lorem",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>,
            <div
              className="text small"
            >
              Ipsum
            </div>,
          ],
          "className": "suggestedItem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>,
                null,
              ],
              "className": "suggestedImageWrapper",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />,
                  "className": "suggestedImage",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "image",
                    "src": "http://via.placeholder.com/300x300",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ipsum",
              "className": "text small",
            },
            "ref": null,
            "rendered": "Ipsum",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>,
            <div
              className="text small"
            >
              Lorem ipsum
            </div>,
          ],
          "className": "suggestedItem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>,
                null,
              ],
              "className": "suggestedImageWrapper",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />,
                  "className": "suggestedImage",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "image",
                    "src": "http://via.placeholder.com/300x300",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem ipsum",
              "className": "text small",
            },
            "ref": null,
            "rendered": "Lorem ipsum",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>,
            <div
              className="text small"
            >
              Sed ut perspiciatis
            </div>,
          ],
          "className": "suggestedItem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>,
                null,
              ],
              "className": "suggestedImageWrapper",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />,
                  "className": "suggestedImage",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "image",
                    "src": "http://via.placeholder.com/300x300",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sed ut perspiciatis",
              "className": "text small",
            },
            "ref": null,
            "rendered": "Sed ut perspiciatis",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="suggestedImageWrapper"
              onClick={[Function]}
            >
              <div
                className="suggestedImage"
              >
                <img
                  className="image"
                  src="http://via.placeholder.com/300x300"
                />
              </div>
            </div>,
            <div
              className="text small"
            >
              perspiciatis
            </div>,
          ],
          "className": "suggestedItem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>,
                null,
              ],
              "className": "suggestedImageWrapper",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />,
                  "className": "suggestedImage",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "image",
                    "src": "http://via.placeholder.com/300x300",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "perspiciatis",
              "className": "text small",
            },
            "ref": null,
            "rendered": "perspiciatis",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input
            doNotScroll={true}
            onChange={[Function]}
            placeholder="Foo"
            searchIcon={true}
            size="small"
          />,
          Array [
            <div
              className="suggestedItem"
            >
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>
              <div
                className="text small"
              >
                Lorem
              </div>
            </div>,
            <div
              className="suggestedItem"
            >
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>
              <div
                className="text small"
              >
                Ipsum
              </div>
            </div>,
            <div
              className="suggestedItem"
            >
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>
              <div
                className="text small"
              >
                Lorem ipsum
              </div>
            </div>,
            <div
              className="suggestedItem"
            >
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>
              <div
                className="text small"
              >
                Sed ut perspiciatis
              </div>
            </div>,
            <div
              className="suggestedItem"
            >
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>
              <div
                className="text small"
              >
                perspiciatis
              </div>
            </div>,
          ],
        ],
        "className": "inputSelectImage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "doNotScroll": true,
            "onChange": [Function],
            "placeholder": "Foo",
            "searchIcon": true,
            "size": "small",
          },
          "ref": "input",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>,
              <div
                className="text small"
              >
                Lorem
              </div>,
            ],
            "className": "suggestedItem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="suggestedImage"
                  >
                    <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />
                  </div>,
                  null,
                ],
                "className": "suggestedImageWrapper",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />,
                    "className": "suggestedImage",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "image",
                      "src": "http://via.placeholder.com/300x300",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem",
                "className": "text small",
              },
              "ref": null,
              "rendered": "Lorem",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>,
              <div
                className="text small"
              >
                Ipsum
              </div>,
            ],
            "className": "suggestedItem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="suggestedImage"
                  >
                    <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />
                  </div>,
                  null,
                ],
                "className": "suggestedImageWrapper",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />,
                    "className": "suggestedImage",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "image",
                      "src": "http://via.placeholder.com/300x300",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ipsum",
                "className": "text small",
              },
              "ref": null,
              "rendered": "Ipsum",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>,
              <div
                className="text small"
              >
                Lorem ipsum
              </div>,
            ],
            "className": "suggestedItem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="suggestedImage"
                  >
                    <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />
                  </div>,
                  null,
                ],
                "className": "suggestedImageWrapper",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />,
                    "className": "suggestedImage",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "image",
                      "src": "http://via.placeholder.com/300x300",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem ipsum",
                "className": "text small",
              },
              "ref": null,
              "rendered": "Lorem ipsum",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>,
              <div
                className="text small"
              >
                Sed ut perspiciatis
              </div>,
            ],
            "className": "suggestedItem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="suggestedImage"
                  >
                    <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />
                  </div>,
                  null,
                ],
                "className": "suggestedImageWrapper",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />,
                    "className": "suggestedImage",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "image",
                      "src": "http://via.placeholder.com/300x300",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sed ut perspiciatis",
                "className": "text small",
              },
              "ref": null,
              "rendered": "Sed ut perspiciatis",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="suggestedImageWrapper"
                onClick={[Function]}
              >
                <div
                  className="suggestedImage"
                >
                  <img
                    className="image"
                    src="http://via.placeholder.com/300x300"
                  />
                </div>
              </div>,
              <div
                className="text small"
              >
                perspiciatis
              </div>,
            ],
            "className": "suggestedItem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="suggestedImage"
                  >
                    <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />
                  </div>,
                  null,
                ],
                "className": "suggestedImageWrapper",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      className="image"
                      src="http://via.placeholder.com/300x300"
                    />,
                    "className": "suggestedImage",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "image",
                      "src": "http://via.placeholder.com/300x300",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "perspiciatis",
                "className": "text small",
              },
              "ref": null,
              "rendered": "perspiciatis",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
